chainer/link.py

```diff
@@ -552,35 +646,57 @@ def __getitem__(self, name):
         """Equivalent to getattr."""
         return getattr(self, name)
 
+    def __setattr__(self, name, value):
```

Comment:
Also, please write to the document that the `name` attribute of `link` registered is overwritten.